network-api/networkapi/wagtailpages/pagemodels/blog/blog.py

```diff
@@ -295,24 +289,24 @@ def get_missing_related_posts(self):
         """
         additional_posts = list()
         post_count = self.related_posts.all().count()
-        missing_count = self.related_post_count - post_count
+        missing_count = self.RELATED_POSTS_MAX - post_count
 
-        if missing_count == 0:
+        if missing_count <= 0:
+            # We have enough related posts already, so return an empty list
             return additional_posts
 
         related_posts = get_content_related_by_tag(self)
 
-        if len(related_posts) > 0:
```

Comment:
The `len(related_posts) > 0` is unnecessary before a for-loop as the for-loop never starts on an empty iterable.